### PR TITLE
Remove unintended dependency on next-auth

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: ?

### What are the changes and their implications?

Remove unintended dependency on next-auth by removing it from the core build of @blitzjs/auth

## Bug Checklist

- [x] Changeset added (run `pnpm changeset` in the root directory)

### CODE REVIEW

File: .changeset/modern-insects-raise.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@blitzjs/auth": patch
+"blitz": patch
+---
+
+Remove unintended dependency on next-auth by updating the build method for next-auth adapter and removing it from the core build of @blitzjs/auth
```
Comment (siddhsuresh) [inline .changeset/modern-insects-raise.md:6]:
```suggestion
Remove unintended dependency on next-auth by removing it from the core build of @blitzjs/auth
```

File: .changeset/modern-insects-raise.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@blitzjs/auth": patch
+"blitz": patch
+---
+
+Remove unintended dependency on next-auth by removing it from the core build of @blitzjs/auth
```
Comment (flybayer) [inline .changeset/modern-insects-raise.md:6]: looks like there is a breaking change for anyone using the adapter, need to document this

**Review by flybayer — APPROVED** [review]

File: .changeset/modern-insects-raise.md (hunk 2 above)
Comment (siddhsuresh) [inline .changeset/modern-insects-raise.md:6]:
```suggestion
Remove unintended dependency on next-auth by removing it from the core build of @blitzjs/auth

⚠️ Breaking Change for current users of `withNextAuthAdapter`

Update your import in `next.config.js` in the following way
```diff
-const { withNextAuthAdapter } = require("@blitzjs/auth")
+const { withNextAuthAdapter } = require("@blitzjs/auth/next-auth")
```
```

File: .changeset/modern-insects-raise.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+⚠️ Breaking Change for current users of `withNextAuthAdapter`
+
+Update your import in `next.config.js` in the following way
+```diff
+-const { withNextAuthAdapter } = require("@blitzjs/auth")
++const { withNextAuthAdapter } = require("@blitzjs/auth/next-auth")
```
Comment (siddhsuresh) [inline .changeset/modern-insects-raise.md:13]:
```suggestion
Remove unintended dependency on next-auth by removing it from the core build of @blitzjs/auth

⚠️ Breaking Change for current users of `withNextAuthAdapter`

Update your import in `next.config.js` in the following way
```diff
-const { withNextAuthAdapter } = require("@blitzjs/auth")
+const { withNextAuthAdapter } = require("@blitzjs/auth/next-auth")
```
```

**Pull request merged**